### PR TITLE
18580 (PR 2) - URL Path Builder (adapters B-C)

### PR DESCRIPTION
**Pull request by mjk90**
## Description
Updated the following adapters to use buildUrlPath: bitex, blockchain.com, blockchair, btc.com, coinapi, coinbase, coincodex, coinmetrics, coinpaprika, covidtracker, cryptoapis, cryptoapis-v2, cryptocompare, cryptomkt
......

## Steps to Test

1. yarn test:unit
2. yarn test:integration
3. Run an adapter with user input included in a url and check the URL it attempts to call.

## Quality Assurance

- [x] Ran `yarn changeset` if adapter source code was changed
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [x] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [x] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)

### CODE REVIEW

**Comment by austinborn** [comment]
I think reviewers will have a hard time reviewing this until the first part is merged. Same for parts 3 and 4.

File: packages/sources/blockchair/src/endpoint/balance.ts [hunk 1]
```diff
@@ -21,10 +21,13 @@ const getBalances: balance.GetBalances = async (accounts, config) => {
 
   const reqConfig: RequestConfig = {
     ...config.api,
-    url: getBalanceURI(addresses, coin as string, chain as string),
+    url: getBalanceURI(coin as string, chain as string),
```
Comment (boxhock) [inline packages/sources/blockchair/src/endpoint/balance.ts:24]: Are these `... as string` necessary? Do we not have types covered for them?

File: packages/sources/blockchair/src/endpoint/balance.ts [hunk 2]
```diff
@@ -21,10 +21,13 @@ const getBalances: balance.GetBalances = async (accounts, config) => {
...
+    params: { addresses: addresses.join(',') },
   }
 
+  console.log({ reqConfig })
```
Comment (boxhock) [inline packages/sources/blockchair/src/endpoint/balance.ts:28]:
```suggestion
```

File: packages/sources/blockchair/src/endpoint/balance.ts [hunk 3]
```diff
@@ -21,10 +21,13 @@ const getBalances: balance.GetBalances = async (accounts, config) => {
...
   const response = await Requester.request(reqConfig)
+  console.log({ response })
```
Comment (boxhock) [inline packages/sources/blockchair/src/endpoint/balance.ts:30]:
```suggestion
```

**Review by boxhock — CHANGES_REQUESTED** [review]

File: packages/sources/blockchair/src/endpoint/balance.ts (hunk 1 above)
Comment (mjk90) [reply packages/sources/blockchair/src/endpoint/balance.ts:24]: `coin` and `chain` come from the `Account` type, which defines them as optional, so if we don't cast them, they cause a typescript error. I could update this method to check if they're valid strings before doing anything else if you think this is an issue

Comment (boxhock) [reply packages/sources/blockchair/src/endpoint/balance.ts:24]: Hmm yeah I think we should do that. If they're not required, I guess we could just do `... || ''` instead?

Comment (austinborn) [reply packages/sources/blockchair/src/endpoint/balance.ts:24]: +1 for `... || ''`, or `... ?? ''` if the type is limited to `string | undefined | null`

**Review by boxhock — APPROVED** [review]

**Pull request merged**